packages/botonic-core/src/core-bot.ts

```diff
@@ -261,4 +270,49 @@ export class CoreBot {
       lastRoutePath: followUp.lastRoutePath,
     }
   }
+
+  private async runFollowUpFlowBuilderBotAction(
+    previousResponse: BotResponse,
+    { input, session, lastRoutePath }: BotRequest
+  ) {
+    const nextPayload = session._botonic_action?.split(
+      'flow_builder_bot_action:'
+    )[1]
+
+    const inputWithBotActionPayload: Input = {
+      ...input,
+      payload: nextPayload,
+      type: INPUT.POSTBACK,
+      data: undefined,
+      text: undefined,
+    }
+
+    session._botonic_action = undefined
+
+    const followUp = await this.runInput({
+      input: inputWithBotActionPayload,
+      session,
+      lastRoutePath,
+    })
+
+    const response = {
+      input: followUp.input,
+      response: previousResponse.response.concat(followUp.response),
+      session,
+      lastRoutePath: followUp.lastRoutePath,
+    }
+
+    // Recursive call to handle multiple bot actions in a row
+    if (
+      response.session._botonic_action?.startsWith('flow_builder_bot_action:')
+    ) {
+      return await this.runFollowUpFlowBuilderBotAction(response, {
```

Comment:
Be careful when using recursive functions. We need to ensure it's not possible to run this infinitely. In the current implementation we're not protected against this (imagine a bot action that is connected to itself, or a chain of bot action 1 -> bot action 2 -> bot action 1). If this happens it's very hard to debug because you just see the process running out of memory or other strange errors.
I suggest adding an additional parameter "n" (by default 0) and in this call doing `return await this.runFollowUpFlowBuilderBotAction(..., n=n+1)` then adding a conditional in the function like `if(n>10) throw new Error('Maximum BotAction recursive calls reached (10)');`